hive/ranger2-client.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/bash
+#
+# @file hive/ranger2-client.sh
+# @brief Provides function to install Apache Ranger-2.0.0 client for Hive
+
+source /usr/lib/qubole/bootstrap-functions/common/utils.sh
+source /usr/lib/qubole/bootstrap-functions/hive/hiveserver2.sh
+
+# @description Install Apache Ranger-2.0.0 client for Hive
+#
+# Currently supported only on AWS
+# Requires HiveServer2
+#
+# @example
+#   install_ranger -h example.host -p 6080 -r examplerepo
+#
+# @arg -h string Hostname of Ranger admin. Defaults to `localhost`
+# @arg -p int Port where Ranger admin is running. Defaults to `6080`
+# @arg -r string Name of Ranger repository. Defaults to `hivedev`
+
+# @arg -S string Hostname of Solr admin. Defaults to `""`
+# @arg -P int Port where Solr admin is running. Defaults to `6083`
+
+function install_ranger() {
+    populate_nodeinfo
+    if [[ is_hs2_configured ]]; then
+        HOST=localhost
+        PORT=6080
+        REPOSITORY_NAME=hivedev
+        QBOL_HEALTH_CHECK_USER=qbol_user
+        SOLR_HOST=""
+        SOLR_PORT=6083
+        
+        while getopts ":h:p:r:S:P" opt; do
+            case ${opt} in
+                h)
+                    HOST=${OPTARG}
+                    ;;
+                p)
+                    PORT=${OPTARG}
+                    ;;
+                r)
+                    REPOSITORY_NAME=${OPTARG}
+                    ;;
+                S)
+                    SOLR_HOST=${OPTARG}
+                    ;;
+                P)
+                    SOLR_PORT=${OPTARG}
+                    ;;                                        
+                \?)
+                    echo "Invalid option: -${OPTARG}" >&2
+                    return 1
+                    ;;
+                :)
+                    echo "Option -${OPTARG} requires an argument." >&2
+                    return 1
+                    ;;
+            esac
+        done
+        
+        URL=http://${HOST}:${PORT}
+        HIVE_LIB=/usr/lib/hive1.2
+	    
+	    S3_PATH=s3://paid-qubole/ranger-2.0.0
+	    RANGER_PLUGIN=ranger-2.0.0-hive-plugin
+    	RANGER_PLUGIN_FILE=${S3_PATH}/${RANGER_PLUGIN}.tar.gz
+  		SPOOL_DIR=/media/ephemeral0/logs/ranger/hive/audit/solr
```

Comment:
Do we need to create this directory in this method itself or does Hive/Ranger take care of creating it ? Same with permission of directory.

Reply:
Ranger will create it.